src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,305 @@
+    oss << "Max Width: " << maxWidth;
+    oss << ", Max Height: " << maxHeight << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+  return status;
+}
+
+bool Image::Load(const std::vector<std::string>& files,
+                 bool flipVertical,
+                 arma::Mat<unsigned char>&& outputMatrix)
+{
+  if (files.size() == 0)
+  {
+    std::ostringstream oss;
+    oss << "File vector is empty." << std::endl;
+    throw std::runtime_error(oss.str());
+  }
+
+  arma::Mat<unsigned char> img;
+  size_t width, height;
+  bool status = Load(files[0], flipVertical, std::move(img),
+                     width, height, channels);
+  Log::Info << "Loaded " << files[0] << std::endl;
+
+  // Decide matrix dimension using the image height and width.
+  maxWidth = std::max(maxWidth, width);
+  maxHeight = std::max(maxHeight, height);
+
+  outputMatrix.set_size(maxWidth*maxHeight*channels, files.size());
+  outputMatrix.col(0) = img;
+
+  for (size_t i = 1; i < files.size() ; i++)
+  {
+    status &= Load(files[i],
+                   flipVertical,
+                   std::move(outputMatrix.col(i)));
+    Log::Info << "Loaded " << files[i] << std::endl;
+  }
+  return status;
+}
+
+
+bool Image::Save(const std::string& fileName,
+                 size_t width,
+                 size_t height,
+                 size_t channels,
+                 bool flipVertical,
+                 arma::Mat<unsigned char>&& inputMatrix)
+{
+  if (!ImageFormatSupported(fileName, true))
+  {
+    std::ostringstream oss;
+    oss << "File type " << Extension(fileName) << " not supported.\n";
+    oss << "Currently it supports ";
+    for (auto extension : saveFileTypes)
+      oss << ", " << extension;
+    oss << std::endl;
+    throw std::runtime_error(oss.str());
+    return false;
+  }
+  if (inputMatrix.n_cols > 1)
+  {
+    std::cout << "Input Matrix contains more than 1 image." << std::endl;
+    std::cout << "Only the firstimage will be saved!" << std::endl;
+  }
+  unsigned char *image = inputMatrix.memptr();
+  stbi_flip_vertically_on_write(flipVertical);
+
+  bool status;
+  int tempWidth = width, tempHeight = height, tempChannels = channels;
+  if ("png" == Extension(fileName))
+  {
+    status = stbi_write_png(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image, width*channels);
+  }
+  else if ("bmp" == Extension(fileName))
+  {
+    status = stbi_write_bmp(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image);
+  }
+  else if ("tga" == Extension(fileName))
+  {
+    status = stbi_write_tga(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image);
+  }
+  else if ("hdr" == Extension(fileName))
+  {
+    status = stbi_write_hdr(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, reinterpret_cast<float*>(image));
+  }
+  else if ("jpg" == Extension(fileName))
+  {
+    status = stbi_write_jpg(fileName.c_str(), tempWidth, tempHeight,
+        tempChannels, image, 95);
```

Comment:
I think we should make the compression configurable.

Reply:
Okay,  I guess we should make it configurable with a default value. Let me know what you think?

Reply:
Agreed, that way we can keep the same interface.